src/lib/marshal/translation/helpers.ts

```diff
@@ -206,6 +210,32 @@ export const ensureValidCommandTarget = async (
   // Got translationRef arg but no --all flag, which means target only a single
   // translation file.
   if (!flags.all) {
+    const isPushOrValidate = [
+      "translation:validate",
+      "translation:push",
+    ].includes(commandId ?? "");
+    if (isPushOrValidate) {
+      // Try to get the translation file for all supported formats when validating or pushing
+      // Start with the specified format then fallback to other options
+      const formatsToLookup = [
+        flags.format,
+        ...SUPPORTED_TRANSLATION_FORMATS,
+      ].filter((x) => Boolean(x));
+
+      let translationFileCtx: TranslationFileContext;
+      for await (const format of formatsToLookup) {
```

Comment:
Continuing from my comment above and below, I think this bit here should flow like this:

```
// If specified, only check the given format; otherwise check both json and po formats.
const formats = if flags.format ? [flags.format] : SUPPORTED_TRANSLATION_FORMATS;

const translationFileCtxs = formats.map(format => buildTranslationFileCtx(...))
if (flags.format) return translationFileCtxs[0];

// If not specified, look for a file that exists in either format and return that file context if found.
const existingFileCtx = translationFileCtxs.find(ctx => ctx.exists)
if (existingFileCtx) return existingFileCtx;

// If none exists, fall back to the file context with a json format (assuming the first one)
return translationFileCtxs[0];
```

Reply:
I added the conditional logic around the command id because on `translation:pull`, if no format was specified, it'd look for `json` and not find it, then look for `po` and not find it, then fallback to the last `translationFileCtx`, so by default it would create the files as json. But, the logic you outlined above bypasses that by getting context for all file types and if neither exist, going with the first which would be json. 

All that being said, totally agree with the logic you outlined above and will update the function.

Reply:
> because on translation:pull, if no format was specified, it'd look for json and not find it, then look for po and not find it, then fallback to the last translationFileCtx

^ to clarify this point, I believe if you don't pass in a `--format` flag manually then [we default to json](https://github.com/knocklabs/knock-cli/blob/main/src/commands/translation/pull.ts#L47) which means we will always have a format set here for the pull command, therefore we will only construct one file context. Lmk if this is not consistent with your understanding!

Reply:
Yeah, we're on the same page. My first pass was just buggy and made that not work well